tests/client/amo/components/TestLandingPage.js

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import {
+  renderIntoDocument,
+  findRenderedComponentWithType,
+} from 'react-addons-test-utils';
+import { findDOMNode } from 'react-dom';
+import { Provider } from 'react-redux';
+
+import * as landingActions from 'amo/actions/landing';
+import {
+  LandingPageBase,
+  mapStateToProps,
+  singularizeAddonType,
+} from 'amo/components/LandingPage';
+import createStore from 'amo/store';
+import I18nProvider from 'core/i18n/Provider';
+import { fakeAddon } from 'tests/client/amo/helpers';
+import { getFakeI18nInst } from 'tests/client/helpers';
+
+
+describe('<LandingPage />', () => {
+  function render({ ...props }) {
+    const initialState = { api: { clientApp: 'android', lang: 'en-GB' } };
+
+    return findDOMNode(findRenderedComponentWithType(renderIntoDocument(
+      <Provider store={createStore(initialState)}>
+        <I18nProvider i18n={getFakeI18nInst()}>
+          <LandingPageBase i18n={getFakeI18nInst()} {...props} />
+        </I18nProvider>
+      </Provider>
+    ), LandingPageBase));
+  }
+
+  it('renders a LandingPage with no addons set', () => {
+    const root = render({
+      addonType: 'extension',
+    });
+
+    assert.include(root.textContent, 'Featured extensions');
+    assert.include(root.textContent, 'More featured extensions');
+  });
+
+  it('renders a LandingPage with themes HTML', () => {
+    const root = render({
+      addonType: 'theme',
+    });
+
+    assert.include(root.textContent, 'Featured themes');
+    assert.include(root.textContent, 'More featured themes');
+  });
+
+  it('renders each add-on when set', () => {
+    const props = {
+      addonType: 'theme',
+      featuredAddons: [
+        { ...fakeAddon, name: 'Howdy', slug: 'howdy' },
+        { ...fakeAddon, name: 'Howdy again', slug: 'howdy-again' },
+      ],
+      highlyRatedAddons: [
+        { ...fakeAddon, name: 'High', slug: 'high' },
+        { ...fakeAddon, name: 'High again', slug: 'high-again' },
+      ],
+      popularAddons: [
+        { ...fakeAddon, name: 'Pop', slug: 'pop' },
+        { ...fakeAddon, name: 'Pop again', slug: 'pop-again' },
+      ],
+    };
+    const root = render(props);
+
+    assert.deepEqual(
+      Object.values(root.querySelectorAll('.SearchResult-heading'))
+        .map((heading) => heading.textContent),
+      ['Howdy', 'Howdy again', 'High', 'High again', 'Pop', 'Pop again']
+    );
+  });
+
+  it('throws if add-on type is not supported', () => {
+    assert.throws(() => {
+      render({ addonType: 'XUL' });
+    }, 'No LandingPage content for addonType: XUL');
+  });
+
+  describe('mapStateToProps()', () => {
```

Comment:
Oh, I get your comment about not deleting much code with this new patch. To be honest, I don't think we need this test as long as you use `store.dispatch()` in the rendering test above. 

We typically isolate state mapper tests from component tests but only for when a real `store` object would make testing the component too difficult. In this case, it doesn't.

To put it another way: if you remove this test you'll still get 100% code coverage because the other rendering test will have to rely on mapStateToProps() to pass.